src/main.rs

```diff
@@ -60,10 +63,49 @@ pub struct UploadResponse {
     url: String,
 }
 
+struct DirEntryModTimePair {
+    dir_entry: DirEntry,
+    mod_time: SystemTime,
+}
+
 async fn index() -> impl Responder {
     HttpResponse::Ok().body("i API ready!")
 }
 
+async fn recent(opt: web::Data<Opt>) -> impl Responder {
```

Comment:
I think if we change the return type to `Result<impl Responder, Error>` we can use the `?` operator on a lot of places, instead of handling each error manuallly with `match`. See below :)

Reply:
Ahh, so that was the reason I couldn't use '?'! This is great. That function looks way better now.